driver-sync/src/test/functional/com/mongodb/client/RetryableWritesProseTest.java

```diff
@@ -246,13 +249,14 @@ public void commandSucceeded(final CommandSucceededEvent event) {
             MongoCollection<Document> collection = client.getDatabase(getDefaultDatabaseName())
                     .getCollection("originalErrorMustBePropagatedIfNoWritesPerformed");
             collection.drop();
-            try {
-                collection.insertOne(new Document());
-            } catch (MongoException e) {
-                assertEquals(e.getCode(), 91);
-                return;
-            }
-            fail("must not reach");
+            assertThrows(MongoWriteConcernException.class, () -> {
+                try {
+                    collection.insertOne(new Document());
+                } catch (MongoWriteConcernException e) {
+                    assertEquals(91, e.getCode());
+                    throw e;
+                }
+            });
```

Comment:
Now that the leak is fixed, we can expect `MongoWriteConcernException` here.

Reply:
FYI this can also be written as:

```
MongoWriteConcernException e = assertThrows(MongoWriteConcernException.class, () ->  collection.insertOne(new Document()));
assertEquals(91, e.getCode());
```

Reply:
Thank you, I have not realized `assertThrows` returns the thrown exception :) Done.